pkg/agent/runner.go

```diff
@@ -194,6 +196,19 @@ func (r *Runner) Run(ctx context.Context, logger *zap.Logger, vmInfoUpdated util
 	pluginRequestJitter := util.NewTimeRange(time.Millisecond, 0, 100).Random()
 
 	coreExecLogger := execLogger.Named("core")
+	clock := logiclock.NewClock(func(duration time.Duration, flag logiclock.Flag) {
+		labels := []string{"false", "false", "false"}
+		if flag.Has(logiclock.Upscale) {
+			labels[0] = "true"
+		}
+		if flag.Has(logiclock.Downscale) {
+			labels[1] = "true"
+		}
+		if flag.Has(logiclock.Immediate) {
+			labels[2] = "true"
+		}
+		r.global.metrics.scalingLatency.WithLabelValues(labels...).Observe(duration.Seconds())
```

Comment:
This seems funky... Is there a reason we can't just have a fixed enum? (if we need to accommodate scaling down & up at the same time, maybe "upscale"/"downscale"/"mixed" ?)

If no, let's have a comment here explaining what's going on?

Reply:
I wanted to have more flexibility to carry more information (e. g. Immediate flag there).

Reply:
I've cleaned it up in 3118e51807e3af62b08bd47b24c02657cd66316c.

Reply:
I still don't quite understand why we can't just use a fixed enum... could you elaborate?

(AFAICT "immediate" is just used to indicate whether a scale-up was requested by vm-monitor... I would remove it for now, potentially adding it back later, if we have a need for it. Maybe in the future it'd make sense to have a "requester" field instead, where it could be "bounds"/"agent"/"monitor"...?)

Reply:
I also don't quite get why the clock needs to care about scaling flags -- feels like it's breaking separation of concerns. But presumably I'm missing something?